motionplan/planManager.go

```diff
@@ -48,6 +49,7 @@ func (pm *planManager) PlanSingleWaypoint(ctx context.Context,
 	seedMap map[string][]referenceframe.Input,
 	goalPos spatialmath.Pose,
 	worldState *referenceframe.WorldState,
+	constraintSpec *pb.Constraints,
```

Comment:
We've been trying to move away from passing around protobuf objects.  I'm wondering if you could move the logic to convert them to constraints to a converison file so that we can immediately parse them before passing them into our planning stack

Reply:
> We've been trying to move away from passing around protobuf objects

This is not actually true; we were at one point, but per James/Eric last I heard we were intending to use protobuf structs anywhere something more complex was not needed.

Reply:
The context for that is messages that are just literally ints/strings, the logic involved with this is significantly more complicated and it would behoove us to type this so that making future changes is not as arduous.  Is there an issue with reframing this as conversion functions?

Reply:
I disagree that the logic here is significantly more complicated. The config is and always has been just some strings/floats, right up until it's turned into the `Constraint` object.

Reply:
No but this message

> per James/Eric last I heard we were intending to use protobuf structs anywhere something more complex was not needed.

refers to messages that are messages that are like `userID string`.  There is more complicated parsing for these messages (they include oneofs, which make them complicated off the bat) and we already have the concepts for `Constraints` within RDK.  What I am proposing is just using `ConstraintsFromProto` at the top level and then passing them in that way.  The motivation being that we don't know how we are going to want to use these in the long term and that we cannot operate on protobuf directly.

Reply:
Stick with the proto. Duplicating the functionality of `pb.Constraint` for no functional gain is not a good use of engineering time.